src/domain/community/contributor/contributor.service.ts

```diff
@@ -15,23 +14,99 @@ import { IAgent } from '@domain/agent/agent/agent.interface';
 import { VirtualContributor } from '../virtual-contributor/virtual.contributor.entity';
 import { CommunityContributorType } from '@common/enums/community.contributor.type';
 import { ContributorLookupService } from '@services/infrastructure/contributor-lookup/contributor.lookup.service';
+import { CreateProfileInput } from '@domain/common/profile/dto';
+import { AvatarCreatorService } from '@services/external/avatar-creator/avatar.creator.service';
+import { IProfile } from '@domain/common/profile/profile.interface';
+import { ProfileService } from '@domain/common/profile/profile.service';
+import { VisualType } from '@common/enums/visual.type';
+import { StorageBucketService } from '@domain/storage/storage-bucket/storage.bucket.service';
+import { AgentInfo } from '@core/authentication.agent.info/agent.info';
+import { DocumentService } from '@domain/storage/document/document.service';
 
 @Injectable()
 export class ContributorService {
   constructor(
     private contributorLookupService: ContributorLookupService,
+    private profileService: ProfileService,
     @InjectEntityManager('default')
-    private entityManager: EntityManager
+    private entityManager: EntityManager,
+    private avatarCreatorService: AvatarCreatorService,
+    private storageBucketService: StorageBucketService,
+    private documentService: DocumentService
   ) {}
 
-  async contributorsWithCredentials(
-    credentialCriteria: CredentialsSearchInput,
-    limit?: number
-  ): Promise<IContributor[]> {
-    return await this.contributorLookupService.contributorsWithCredentials(
-      credentialCriteria,
-      limit
+  public async addAvatarVisualToContributorProfile(
+    profile: IProfile,
+    profileData: CreateProfileInput,
+    agentInfo?: AgentInfo,
+    firstName?: string,
+    lastName?: string
+  ): Promise<void> {
+    let avatarURL: string | undefined = undefined;
+    if (profileData.avatarURL && profileData.avatarURL !== '') {
```

Comment:
`&& profileData.avatarURL !== ''` is redundant

Reply:
the alternative is to check that it is an actual URL, which may be the better check. But we do need to guard that there is a meaningful avatar URL given here